Task #107 limit page view by role

strikeyamato opened the pull request:
Changes done to the backend and frontend in order to implement Role Base Access Control (RBAC). 

To note, I have not configured the code to access the new host so please keep that in mind when reviewing this PR

strikeyamato committed:
Refactor Part one: central_dashboard + /views.py update

Began refactoring process for task #107 starting with creating a new Django app "central_dashboard". Updated it's views.py file to account for role permissions when accessing views.

To Do:
- Add serializers.py and urls.py files.
- Add tests to tests.py
- Modify frontend files to be compatible with new logic for accessing different functions (example, only admins and managers should be able to add or remove a part from the Inventory page).

strikeyamato committed:
Refactor Part two: serializer.py, urls.py, tests.py

Added the serializer.py and urls.py files to centralized_dashboard.

Added 9 tests to tests.py and all are OK.

To Do:
- Begin refactoring frontend to accomodate new changes to backend.
- Update API endpoints and ensure functionality
- Add tests to frontend that account for changes in permissions logic
- remove admin_dashboard + manager_dashboard once refactoring is complete

strikeyamato committed:
Revert "Refactor Part two: serializer.py, urls.py, tests.py"

This reverts commit 1a8d1ace00a97e3086e9520bdb766c26dab5c5f4.

strikeyamato committed:
Revert "Refactor Part one: central_dashboard + /views.py update"

This reverts commit f1ab82c9046afcf9cd2ee6221ef1d444750af273.

strikeyamato committed:
Merge branch 'main' into task-107-limit-page-view-by-role

strikeyamato committed:
Refactor part 3: Reverting changes, implementing new ones

Reverted original two commits for task #107 as I misunderstood the requirement: Multiple dashboards will be created for each user role and certain pages will only be accessible on certain dashboards.

Implemented role verification at the dashboard level for AdminDashboard.

Also added role verification in AddUserDashboard.

To Do:
- Continue modifying the rest of the admin_dashboard frontend files to incorporate role verification

- Add tests

- Possibly refactor dashboard_routes.jsx so that role verification need only occur once at the dashboard level (AdminDashboard would be a parent to other elements such as ManageUsersPage)

strikeyamato committed:
Update dashboard_routes.jsx

#107 Updated the dashboard_routes to have admin only elements be children of the Admin Dashboard frontend. This way, role verification need only be done once in Admin_dashboard.jsx.

To Do:
- Add tests to ensure functionality
- Confirm if there are any missing paths that need to be directed to the admin dashboard

strikeyamato committed:
Merge branch 'main' into task-107-limit-page-view-by-role

strikeyamato committed:
Merge branch 'main' into task-107-limit-page-view-by-role

strikeyamato committed:
Merge branch 'main' into task-107-limit-page-view-by-role

strikeyamato committed:
Merge branch 'main' into task-107-limit-page-view-by-role

strikeyamato committed:
Backend refactoring

#107

To ensure role verification is implemented correctly, it must be done in the backend instead of the frontend. Began that process.

To Do: Finish backend refactoring, start/finish frontend

strikeyamato committed:
View Removal

#107 Initially added some additional models + views but have now removed them since they are not needed.

To Do:
- Complete adding Role verification to backend.
- Make frontend changes to be compatible with refactored role identification

strikeyamato committed:
Backend 90% Refactored #107

Backend is refactored but I'm still running into issues with being unable to prevent rerouting by simply changing URL.

To Do:
- Fix issues with Backend Authentication
- Refactor frontend

strikeyamato committed:
#107 Protect Routes implemented

Created a protected_routes file that prevents navigation to a url if the user has an unauthorized role. Paths are subject to change in future iterations as required.

dashboard_routes updated to use protected_routes.

To Do: in future iteration: Refactor paths to have structure such as
('/parent' +
'/child1'
'/child2'
'/etc')

strikeyamato committed:
Merge branch 'main' into task-107-limit-page-view-by-role

sarahabellard reviewed (changes requested):
The CI run is failing, please fix

strikeyamato committed:
Merge branch 'main' into task-107-limit-page-view-by-role

m-coscia committed:
fix message causing error

strikeyamato committed:
#107 Updated Tests

Updated Inventory and ManufacturingLists tests to comply with backend role based authentication.

Consequentently I had to also update inventory view to remove the "safe=False" parameter as it is incompatible (generating a failure when runnning tests)

m-coscia committed:
Merge branch 'main' into task-107-limit-page-view-by-role

m-coscia reviewed (approved):
Significant difference from before, good work
